tests/cupy_tests/logic_tests/test_comparison.py

```diff
@@ -74,6 +74,27 @@ def test_binary_array_pyscalar(self, xp, dtype):
         return [op(a, b) for op in self.operators]
 
 
+class TestArrayEqual(unittest.TestCase):
+
+    @testing.numpy_cupy_equal()
+    def test_array_equal_not_equal(self, xp):
+        a = xp.array([1, 2, 3, 4])
+        b = xp.array([0, 0, 1000, 1000])
+        return xp.allclose(a, b)
+
+    @testing.numpy_cupy_equal()
+    def test_array_equal_is_equal(self, xp):
+        a = xp.array([1, 2, 3, 4])
+        b = xp.array([1, 2, 3, 4])
+        return xp.allclose(a, b)
+
+    @testing.numpy_cupy_equal()
+    def test_array_equal_diff_length(self, xp):
+        a = xp.array([1, 2, 3, 4])
+        b = xp.array([1, 2, 3])
+        return xp.allclose(a, b)
+
```

Comment:
Please add tests check
- broadcasts are not allowed
- comparison with ndarrays of different dtypes is allowed

Reply:
Why is allclose called instead of array_equal?

Reply:
Oh, I actually copy pasted the tests from `allclose` for my convenience and mistakenly left that call unchanged. I think I should've checked it..

I'll edit that soon.